haskell.nix/.buildkite/pipeline.yml

```diff
@@ -0,0 +1,28 @@
+steps:
+
+# build haskell components
+- label: build
+  commands:
+  - nix-build -A checks.x86_64-linux.build-all
+
+# don't run tests until all components are built
+- wait
+
+# run the tests
+- label: test
+  commands:
+  - nix-build -A checks.x86_64-linux.test
```

Comment:
We can actually use `nix flake check` now on our Buildkite

Reply:
I think using `nix-build` here is better so we don't have to worry about bugs emerging in `nix flake`, or the interface changing. I know `nix flake check` has some extra checks for the general flake validness, but I think they are not important for the CI.

Reply:
I guess understand wanting to avoid interface changes and stick with something more stable (although it seems like that will most likely remain the same) but I still disagree with this for 2 reasons.

I don't like how explicit the command is, as it includes both the name of the check and the name of the platform, which makes updating it alongside other changes more necessary rather than something which is likely to adapt. Also I think that checking flake validity is a good thing, while the goal of this template (which I still disagree with) seems to be just to utilize Nix for reproducible CI for Haskell code, these Nix definitions will become a component of the project once included, so CI checking that too also makes sense, plus it in theory helps ensure that the Haskell CI is implemented and functioning correctly.

Reply:
> I don't like how explicit the command is, as it includes both the name of the check and the name of the platform, which makes updating it alongside other changes more necessary rather than something which is likely to adapt.

I actually think this explicitness is better to people not super-familiar with nix, otherwise it may be confusing to them why we specify system in some places but not in others.

I'm not sure what kind of updates you mean that require too many changes.

> Also I think that checking flake validity is a good thing

Again, I'm not sure which checks you mean and in which situations they would help us. To me it does not seem like there is anything useful for us, that would be worth sacrificing some portion of CI stability.

Reply:
What we can do is try to add a separate buildkite/gitlab job just for the flake validness. Though I don't see how it would be useful, except for situations where we make a flake which also needs to be used externally, not only in CI, which we would almost never need for our haskell projects. So it seems unnecessary for the basic template we have here.